src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,81 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [loaded, setLoaded] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const { userId, id, title, completed } = currentTodo as Todo;
+
+  useEffect(() => {
+    const loadUser = async () => {
+      const loadedUser = getUser(userId);
```

Comment:
The `getUser` function call should be awaited directly to ensure that the user data is fetched before proceeding. Change `const loadedUser = getUser(userId);` to `const loadedUser = await getUser(userId);`.